common/utils/ClockTest.cpp

```diff
@@ -162,11 +208,65 @@ void ClockTest::testClock() {
   OLA_ASSERT_LT(first, second);
 }
 
+/**
+ * @brief Test the mock monotonic clock
+ * 
+ */
+void ClockTest::testMockClockMonotonic() {
+  MockClock clock;
+
+  TimeStamp first;
+  clock.CurrentMonotonicTime(&first);
+
+  TimeInterval one_second(1, 0);
+  clock.AdvanceTime(one_second);
+
+  TimeStamp second;
+  clock.CurrentMonotonicTime(&second);
+  OLA_ASSERT_LT(first, second);
+  OLA_ASSERT_TRUE(one_second <= (second - first));
+
+  TimeInterval ten_point_five_seconds(10, 500000);
+  clock.AdvanceTime(10, 500000);
+
+  TimeStamp third;
+  clock.CurrentMonotonicTime(&third);
+  OLA_ASSERT_LT(second, third);
+  OLA_ASSERT_TRUE(ten_point_five_seconds <= (third - second));
+}
+
+/**
+ * @brief Test the mock real time clock
+ * 
+ */
+void ClockTest::testMockClockRealTime() {
+  MockClock clock;
+
+  TimeStamp first;
+  clock.CurrentRealTime(&first);
+
+  TimeInterval one_second(1, 0);
+  clock.AdvanceTime(one_second);
+
+  TimeStamp second;
+  clock.CurrentRealTime(&second);
+  OLA_ASSERT_LT(first, second);
+  OLA_ASSERT_TRUE(one_second <= (second - first));
```

Comment:
This was a copy-paste of an existing clock unit test and perhaps I'm not seeing what the issue is yet though. Could you elaborate on why this would fail?

Reply:
Well the whole reason for switching to monotonic is because real time may move backwards or forwards due to outside influences. Consider the following:

1. `clock.CurrentRealTime(&first);` 00:00:10
1. `clock.AdvanceTime(one_second);` 00:00:11
1. NTP causes wall clock to roll back 2s due to a leap seconds (it got really bad so we added two!)
1. `clock.CurrentRealTime(&second);` 00:00:09
1. `OLA_ASSERT_LT(first, second);` - This is now false
1. `OLA_ASSERT_TRUE(one_second <= (second - first));` 1 <= 9-10 == 1 <= -1 - This is also now false

It's probably worth switching these to the _MSG variants, or adding one here for the non-existent LT_MSG so we can give the user a little message in the unlikely event the test fails:
https://github.com/OpenLightingProject/ola/blob/0.10/include/ola/testing/TestUtils.h

Reply:
This is still outstanding @stefansiegfried .

Reply:
Great explanation, that is a valid point for real clock. My confusion was that the line numbers referenced (253-254) are actually part of the monotonic clock test and in this case I don't believe time can roll back.

Reply:
Yeah I don't know what's going on there. I think those numbers are stale. If you look here on the files page, it shows I am within testMockClockRealTime. This caught me out earlier too:
https://github.com/OpenLightingProject/ola/pull/1673/files#diff-b8f53e67619b3b3cb1ef5e4f7011f74d3cfdb1959f4b0e94e6f2d53a1ee82062

Reply:
Done.